ui/apps/platform/src/Containers/Policies/Wizard/Step3/BooleanPolicyLogicSection.tsx

```diff
@@ -23,21 +20,22 @@ type BooleanPolicyLogicSectionProps = {
 function BooleanPolicyLogicSection({ readOnly = false }: BooleanPolicyLogicSectionProps) {
     const [descriptor, setDescriptor] = React.useState<Descriptor[]>([]);
     const { values } = useFormikContext<Policy>();
+    const { isFeatureFlagEnabled } = useFeatureFlags();
 
-    const isImageSigningEnabled = useFeatureFlagEnabled('ROX_VERIFY_IMAGE_SIGNATURE');
-    const isNetworkPolicyFieldsEnabled = useFeatureFlagEnabled('ROX_NETPOL_FIELDS');
     React.useEffect(() => {
-        if (values.eventSource === 'AUDIT_LOG_EVENT') {
-            setDescriptor(auditLogDescriptor);
-        } else {
-            setDescriptor([
-                ...policyConfigurationDescriptor,
-                ...networkDetectionDescriptor,
-                ...(isImageSigningEnabled ? [imageSigningCriteriaDescriptor] : []),
-                ...(isNetworkPolicyFieldsEnabled ? networkPolicyFieldDescriptors : []),
-            ]);
-        }
-    }, [values.eventSource, isImageSigningEnabled, isNetworkPolicyFieldsEnabled]);
+        const unfilteredDescriptors =
+            values.eventSource === 'AUDIT_LOG_EVENT'
+                ? auditLogDescriptor
+                : policyConfigurationDescriptor;
+        setDescriptor(
+            unfilteredDescriptors.filter((unfilteredDescriptor) => {
+                if (typeof unfilteredDescriptor.featureFlagDependency === 'string') {
+                    return isFeatureFlagEnabled(unfilteredDescriptor.featureFlagDependency);
+                }
+                return true;
+            })
+        );
+    }, [values.eventSource]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
@alwayshooin Did we discuss one time whether `descriptor` can be a derived variable?

Something like:

```ts
    const unfilteredDescriptors =
        values.eventSource === 'AUDIT_LOG_EVENT'
            ? auditLogDescriptor
            : policyConfigurationDescriptor;
    const descriptor = unfilteredDescriptors.filter((unfilteredDescriptor) => {
        if (typeof unfilteredDescriptor.featureFlagDependency === 'string') {
            return isFeatureFlagEnabled(unfilteredDescriptor.featureFlagDependency);
        }
        return true;
    });
```